cmd/otelcol/main_windows.go

```diff
@@ -21,40 +21,41 @@ package main
 import (
 	"fmt"
 	"os"
+	"syscall"
 
 	"go.opentelemetry.io/collector/otelcol"
+	"golang.org/x/sys/windows"
 	"golang.org/x/sys/windows/svc"
 )
 
 func run(params otelcol.CollectorSettings) error {
-	if useInteractiveMode, err := checkUseInteractiveMode(); err != nil {
-		return err
-	} else if useInteractiveMode {
-		return runInteractive(params)
-	} else {
-		return runService(params)
-	}
-}
-
-func checkUseInteractiveMode() (bool, error) {
-	// If environment variable NO_WINDOWS_SERVICE is set with any value other
-	// than 0, use interactive mode instead of running as a service. This should
-	// be set in case running as a service is not possible or desired even
-	// though the current session is not detected to be interactive
```

Comment:
FWIW this code is copied from https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/builder/internal/builder/templates/main_windows.go.tmpl#L29